storage/src/main/kotlin/InMemoryUserRepository.kt

```diff
@@ -0,0 +1,59 @@
+/**
+ * In-memory implementation of the UserRepository interface.
+ */
+class InMemoryUserRepository : UserRepository {
+
+    private val users = mutableMapOf<String, User>()
+
+    /**
+     * Saves a given user.
+     * @param user the user to save
+     * @return the saved user with a generated ID
+     */
+    override fun save(user: User): User {
+        val userId = java.util.UUID.randomUUID().toString()
+        val newUser = user.copy(id = userId)
+        users[userId] = newUser
+        return newUser
+    }
+
+    /**
+     * Retrieves a user by their ID.
+     * @param userId the ID of the user to retrieve
+     * @return the user with the given ID, or null if no user found
+     */
+    override fun findById(userId: String): User? {
+        return users[userId]
+    }
+
+    /**
+     * Updates a given user.
+     * @param user the user to update
+     * @return the updated user, or null if the user does not exist
+     */
+    override fun update(user: User): User? {
+        return if (users.containsKey(user.id)) {
+            users[user.id] = user
+            user
+        } else {
+            null
+        }
+    }
```

Comment:
I think you could improve with

```suggestion
    override fun update(user: User): User? =
    	users.takeIf { it.containsKey(user.id) }
    	    ?.apply { this[user.id] = user}
    	    ?.let { user }
```